src/main/java/bio/terra/service/filedata/azure/util/BlobContainerClientFactory.java

```diff
@@ -0,0 +1,187 @@
+package bio.terra.service.filedata.azure.util;
+
+import com.azure.core.credential.AzureSasCredential;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.HttpClient;
+import com.azure.storage.blob.*;
+import com.azure.storage.blob.models.UserDelegationKey;
+import com.azure.storage.blob.sas.BlobSasPermission;
+import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import com.azure.storage.common.sas.SasProtocol;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.Locale;
+import java.util.Objects;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * A class that wraps the SDK container client and facilities the creation of SAS Urls for blobs by
+ * abstracting the different strategies that depend on the authentication mechanism.
+ */
+public class BlobContainerClientFactory {
+
+  public static final int EXPIRATION_DAYS = 7;
+  private static final HttpClient httpClient = HttpClient.createDefault();
+
+  private final BlobContainerClient blobContainerClient;
+
+  private final SasTokenGeneratorStrategy sasTokenGeneratorStrategy;
+  private final AzureSasCredential blobContainerSasTokenCreds;
+  private BlobServiceClient blobServiceClient;
```

Comment:
It looks like this field can be `final`. To make the code clearer and easier to read, all fields that can be final should be declared `final`. To make code easier to understand and maintain, classes should be written so that all fields that are effectively final -- their state doesn't changed over the lifetime of the object -- are in fact `final`.

Reply:
Agreed. It should be final.

Reply:
Fixed.